packages/block-editor/src/components/media-replace-flow/index.js

```diff
@@ -134,8 +154,13 @@ const MediaReplaceFlow = ( {
 				<>
 					<NavigableMenu className="block-editor-media-replace-flow__media-upload-menu">
 						<MediaUpload
-							value={ mediaId }
-							onSelect={ ( media ) => selectMedia( media ) }
+							gallery={ multiple && onlyAllowsImages() }
```

Comment:
It looks like this gets re-calculated on every render. I'm not sure if it's really needed as an optimisation, but if the value changes only when `allowedTypes` changes, could we replace the definition of `onlyAllowsImages` with a `useMemo` call and store the result in a const?

Reply:
I don't think there was a performance issue here, but I have moved the function call out of the prop just to be sure. Because `onlyAllowsImages` is inexpensive and returns a boolean I didn't think useMemo was needed though?

Reply:
It's probably not needed — I think I just had `useMemo` on my brain after looking @aaronrobertshaw's recent efforts on improving performance in the `ToolsPanel`. Just a good one for us to keep in the back of our minds in case we want to optimise anything further down the track. Looks good to me!